app/views/proposals/_booking.html.erb

```diff
@@ -0,0 +1,38 @@
+<tr>
+  <td><%= link_to booking.space.title, space_path(booking.space) %></td>
+  <td><%= l booking.date_from, format: :short %></td>
+  <td><%= l booking.date_until, format: :short %></td>
+  <% renter = booking.user %>
+  <td><%= link_to renter.username, user_path(renter) %></td>
+  <td class="text-right">
+  <% if state == 'accepted' || state == 'pending' %>
+    <% if renter.phone_number %>
+    <div class="btn btn-default btn-xs" do %>
+      <span class="glyphicon glyphicon-earphone" data-toggle="tooltip" data-placement="top" title="Call to <%= renter.first_name %> at <%= renter.phone_number%>"></span>
+    </div>
+    <% end %>
+    <%= mail_to renter.email,
+                class: "btn btn-info btn-xs" do %>
+      <span class="glyphicon glyphicon-envelope" data-toggle="tooltip" data-placement="top" title="Send an email to <%= renter.first_name %>"></span>
+    <% end %>
+  <% end %>
+  <% if state == 'sent' %>
+    <%= link_to aprove_proposal_path(booking),
+                class: "btn btn-primary btn-xs" do %>
+      <span class="glyphicon glyphicon-ok"> Aprove</span>
+    <% end %>
+  <% elsif state == 'pending' %>
+    <%= link_to accept_proposal_path(booking),
+                class: "btn btn-success btn-xs" do %>
+      <span class="glyphicon glyphicon-ok"> Accept</span>
+    <% end %>
+  <% end %>
+  <% if state == 'sent' || state == 'pending' %>
```

Comment:
In my opinion, you must move this button for a partial and call it on below conditions.